lnrpc/chainrpc/chainnotifer_server.go

```diff
@@ -196,14 +196,26 @@ func (s *Server) RegisterWithRootServer(grpcServer *grpc.Server) error {
 func (s *Server) RegisterConfirmationsNtfn(in *ConfRequest,
 	confStream ChainNotifier_RegisterConfirmationsNtfnServer) error {
 
-	// We'll start by reconstructing the RPC request into what the
-	// underlying ChainNotifier expects.
+	// A height hint is required to prevent scanning the chain from genesis.
+	if in.HeightHint == 0 {
+		return errors.New("a height hint must be provided")
+	}
+
+	// If the number of confirmations isn't specified, we'll use a sane
+	// default of 1.
+	numConfs := in.NumConfs
+	if numConfs == 0 {
+		numConfs = 1
+	}
```

Comment:
I think we should return an error here too. Keep the api as strict as we can. We need to educate the callers, not camouflage potential bugs.

Reply:
Done.